src/lib/sidenav/drawer.ts

```diff
@@ -675,16 +680,20 @@ export class MatDrawerContainer implements AfterContentInit, OnDestroy {
   _closeModalDrawer() {
     // Close all open drawers where closing is not disabled and the mode is not `side`.
     [this._start, this._end]
-      .filter(drawer => drawer && !drawer.disableClose && drawer.mode !== 'side')
+      .filter(drawer => drawer && !drawer.disableClose && this._canHaveBackdrop(drawer))
       .forEach(drawer => drawer!.close());
   }
 
   _isShowingBackdrop(): boolean {
-    return (this._isDrawerOpen(this._start) && this._start!.mode != 'side')
-        || (this._isDrawerOpen(this._end) && this._end!.mode != 'side');
+    return (this._isDrawerOpen(this._start) && this._canHaveBackdrop(this._start)) ||
+           (this._isDrawerOpen(this._end) && this._canHaveBackdrop(this._end));
+  }
+
+  private _canHaveBackdrop(drawer: MatDrawer): boolean {
+    return drawer.mode !== 'side' || !!this._backdropOverride;
   }
 
-  private _isDrawerOpen(drawer: MatDrawer | null): boolean {
+  private _isDrawerOpen(drawer: MatDrawer | null): drawer is MatDrawer {
```

Comment:
is this a mistake or some crazy typescript thing I don't know about?

Reply:
It's a crazy TS thing. It lets TS know that from now on the drawer is going to be a `MatDrawer` rather than a `MatDrawer | null`.

Reply:
very cool